packages/neuron-ui/src/components/MultisigAddress/hooks.ts

```diff
@@ -26,90 +27,111 @@ export const useSearch = (clearSelected: () => void) => {
+    ({ m, n, r, addresses }: { m: number; n: number; r: number; addresses: string[] }) => {
       return saveMultisigConfig({
         m,
         n,
         r,
-        addresses,
-        fullPayload,
+        blake160s: addresses.map(v => addressToScript(v).args),
         walletId,
       }).then(res => {
         if (isSuccessResponse(res)) {
-          if (res.result) {
-            setConfigs(v => [res.result!, ...v])
-          }
+          setEntities(v => (res.result ? [res.result, ...v] : v))
         } else {
           throw new Error(typeof res.message === 'string' ? res.message : res.message.content)
         }
       })
     },
-    [walletId, setConfigs]
+    [walletId, setEntities]
   )
   useEffect(() => {
-    getMultisigConfig({
-      walletId,
-    }).then(res => {
-      if (isSuccessResponse(res)) {
-        setConfigs(res.result)
+    getMultisigConfig(walletId).then(res => {
+      if (isSuccessResponse(res) && res.result) {
+        setEntities(res.result)
       }
     })
-  }, [setConfigs, walletId])
+  }, [setEntities, walletId, isMainnet])
   const updateConfig = useCallback(
     (id: number) => (alias: string | undefined) => {
       updateMultisigConfig({ id, alias: alias || '' }).then(res => {
         if (isSuccessResponse(res)) {
-          setConfigs(v => v.map(config => (config.id === res.result?.id ? res.result : config)))
+          setEntities(v => v.map(config => (res.result && config.id === res.result?.id ? res.result : config)))
         }
       })
     },
-    [setConfigs]
+    [setEntities]
   )
-  const filterConfig = useCallback((key: string) => {
-    setConfigs(v =>
-      v.filter(config => {
-        return config.alias?.includes(key) || config.fullPayload === key
-      })
-    )
-  }, [])
   const deleteConfigById = useCallback(
     (id: number) => {
-      setConfigs(v => v.filter(config => config.id !== id))
+      setEntities(v => v.filter(config => config.id !== id))
     },
-    [setConfigs]
+    [setEntities]
   )
   const onImportConfig = useCallback(() => {
-    importMultisigConfig({ isMainnet, walletId }).then(res => {
+    importMultisigConfig(walletId).then(res => {
       if (isSuccessResponse(res) && res.result) {
         const { result } = res
         if (result) {
-          setConfigs(v => [...result, ...v])
+          setEntities(v => [...result, ...v])
         }
       }
     })
-  }, [walletId, isMainnet])
+  }, [walletId])
+  const [searchKeywords, setSearchKeywords] = useState('')
+  const onFilterConfig = useCallback(
+    (v: string) => {
+      setSearchKeywords(v)
+    },
+    [setSearchKeywords]
+  )
+  const allConfigs = useMemo<MultisigConfig[]>(
+    () =>
+      entities.map(entity => ({
+        ...entity,
+        addresses: entity.blake160s.map(args =>
+          scriptToAddress(
+            {
+              args,
+              codeHash: DefaultLockInfo.CodeHash,
+              hashType: DefaultLockInfo.HashType,
+            },
```

Comment:
Have we verified addresses on importing configs that they are all `secp256k1 address`s

Reply:
It has verified at before steps.